docs/enhancements/pending/EP-0028-phase-1-scoring-api.md

```diff
@@ -0,0 +1,1208 @@
+4. **Training Data**: Identify high-quality sessions for future model fine-tuning
+
+### Solution Approach
+
+Implement an LLM-based "judge" that critically evaluates completed alert analysis sessions using a rigorous scoring framework. The judge examines investigation methodology, tool usage, evidence gathering, and conclusion quality—providing both quantitative scores and actionable improvement suggestions.
+
+## Overview
+
+This EP describes Phase 1 of the alert session scoring system: implementing a core API for scoring individual TARSy alert analysis sessions using an LLM-based "judge" that critically evaluates analysis quality.
+
+**Scope:** On-demand scoring API for individual sessions. UI integration, scheduled scoring, and insights aggregation are covered in separate future EPs.
+
+## Design Overview
+
+### Architecture Summary
+
+The scoring system introduces a new service layer that operates independently of alert processing. It retrieves completed session data from the existing History Service, sends it to a configurable judge LLM for evaluation, and stores structured scores in a purpose-built database schema.
+
+### Key Design Principles
+
+- **Non-Intrusive**: Operates post-session; zero impact on alert processing performance
+- **Criteria Evolution**: Content-addressed configuration enables tracking scoring criteria changes over time
+- **Flexible Scoring**: JSONB score breakdowns adapt to evolving evaluation dimensions without schema migrations
+- **Actionable Feedback**: Normalized storage of missing tools and alternative approaches enables analytics
+- **Manual Control**: Phase 1 is operator-triggered only; automation deferred to future phases
+
+### Design Goals
+
+- Enable systematic quality assessment of TARSy investigation sessions
+- Provide actionable feedback for improving investigation methodology
+- Establish foundation for agent improvement and MCP server development priorities
+- Support historical analysis and criteria evolution tracking
+- Maintain scoring consistency while allowing criteria iteration
+
+## Experimental Background
+
+**Note**: Initial prototype exists at https://github.com/metlos/TARSy-response-score (Go implementation). This EP proposes native integration into TARSy backend.
+
+**Prototype Findings**:
+- Judge LLM provides critical scoring even for self-produced analyses
+- Input: Complete session conversation from `/final-analysis` endpoint (~25k tokens)
+- Output: JSON score report (~6-7KB)
+- Context window: Comfortably fits in 1M token window (Pro models) or 32k window (Flash models)
+- Scoring consistency: Reliable results across multiple test sessions
+
+## Goals
+
+- Provide API endpoints for scoring individual alert analysis sessions
+- Store scores persistently in database
+- Enable manual quality assessment of TARSy analyses
+- Establish foundation for future UI integration and automation
+
+## Non-Goals (Future Phases)
+
+- UI integration for displaying scores
```

Comment:
Well... I would include some simple/primitive UI in Phase 1. Otherwise we would have to use API any time we want to see the score/analysis. So we would have to build something anyway.